src/group.cr

```diff
@@ -0,0 +1,184 @@
+require "c/grp"
+require "c/sys/limits"
+
+class Group
+  # Converts group ID into a groupname.
+  #
+  # Returns: The groupname for the given group ID.
+  #
+  # ```
+  # Group.name(0) # => root
+  # ```
+  def self.name(gid : Int) : String
+    group_struct = LibC.getgrgid(int_to_gid(gid))
+    return String.new(group_struct.value.gr_name) if !group_struct.null?
+    raise NotFound.new("Group with gid '#{gid}', was not found.")
+  end
+
+  # Converts groupname into group ID.
+  #
+  # Returns: The group ID for the given groupname.
+  # Raises: Group::NotFound error if no group exists with the given groupname.
+  #
+  # ```
+  # Group.gid("root") # => 0
+  # ```
+  def self.gid(group : String | Symbol) : Int
+    group_struct = LibC.getgrnam(group.to_s.check_no_null_byte)
+    return group_struct.value.gr_gid if !group_struct.null?
+    raise NotFound.new("Group with name '#{group.to_s}', was not found.")
+  end
+
+  # Checks to see if a group with the given group ID exists.
+  #
+  # Returns: `true` if the group exists.
+  #
+  # ```
+  # Group.exists?("root")              # => true
+  # Group.exists?("nonexistant_group") # => false
+  # ```
+  def self.exists?(gid : Int) : Bool
+    return !LibC.getgrgid(int_to_gid(gid)).null?
+  end
+
+  # Checks to see if group with a given groupname exists.
+  #
+  # Returns: `true` if the group exists.
+  #
+  # ```
+  # Group.exists?(0)     # => true
+  # Group.exists?(32766) # => false
+  # ```
+  def self.exists?(group : String | Symbol) : Bool
+    return !LibC.getgrnam(group.to_s.check_no_null_byte).null?
+  end
+
+  # Returns the group specified by the group ID.
+  # ```
+  # Group[0]
+  # ```
+  def self.[](gid : Int) : Group
```

Comment:
These methods are aliases of `new`. In Crystal we've decided that we don't want to add method aliases, since all developers end up learning both ones either way (it just creates conflict), so can you remove these?

Reply:
I stand by this use even if it is an alias of `new`. The reason being is I don't think `new` is a good way of looking at 'creating' a user. Using square brackets is more descriptive of what it is, a query.

```
User.new(:root)
```
Reads like you are creating a new user named root.

```
User[:root]
```
Reads like you are asking for the root user which is really what you are doing.

Any other thoughts?

Reply:
I understand your concern, but I I'm not sure to like `User["root"]` either. Maybe something more explicit, like: 
```crystal
struct System::User
  def self.get(name)
    new(name)
  end

  private def initialize(name)
  end
end

user = System::User.get(name)
```

Reply:
This was why I suggested using Process.user instead of User.new. I think I agree on process being the wrong module, we already have a system module, no?

Reply:
To make that more clear, I'm suggesting using System.user as a raising method and System.user? as a nilable query.

Reply:
@RX14 - Sorry about the negative but I just **really** don't like that Idea.

@ysbaddaden - I can get behind a `User.get(:root)` or `User.fetch("root")` but I still like the simplicity of `User[:root]` as its nice and clean and array like.

Reply:
That's the problem: a type isn't an array. This construct feels weird in crystal.

I like the solution by @RX14 with a raising `System.user(name)` and a question marked version that returns a nilable. This scheme is a convention in the Crystal ecosystem and should be used here.

Reply:
I'm afraid `System.user(:root)` is only nice when used as is. Since the name should be configurable it will be a string, and `System.user(name)` supporting a `String|Symbol` union isn't useful.

Reply:
I think I have solved this in a way that is amenable to all of us. 
- I have removed `User[]` and `Group[]`.
- I have added `User.get()` and `Group.get()`.
- I have added `System.user()` and `System.group()`.
- I have moved `User` and `Group` to `System::User` and `System::Group`.

This way you can choose to go through the `User`/`Group` class or `System`.

Reply:
In Crystal we don't have aliases - we have one way to do things. Having these two methods just means every developer has to learn about both methods anyway (since they read other's code), and that there's pointless conflict and inconsistency between developers who prefer one and developers who prefer the other.